File: 向量检索原理/度量方法算法实现/manhattan_distance.cpp
// manhattan_distance.cpp
#include <vector>
#include <cstdlib>
#include <iostream>
#include <stdexcept>
#include <cmath>
double manhattan_distance(const std::vector<double>& vec1, const std::vector<double>& vec2) {
    if (vec1.size() != vec2.size()) throw std::invalid_argument("Vectors must have the same length.");
    double sum = 0.0;
    for (size_t i = 0; i < vec1.size(); ++i) {
        sum += std::abs(vec1[i] - vec2[i]);
    }
    return sum;
}

int main() {
    std::vector<double> vec1 = {1.0, 2.0, 3.0};
    std::vector<double> vec2 = {4.0, 5.0, 6.0};
    std::cout << "Manhattan Distance: " << manhattan_distance(vec1, vec2) << std::endl;
    return 0;
}

File: 向量检索原理/度量方法算法实现/manhattan_distance.py
# manhattan_distance.py
def manhattan_distance(vec1, vec2):
    if len(vec1) != len(vec2):
        raise ValueError("Vectors must have the same length.")
    return sum(abs(a - b) for a, b in zip(vec1, vec2))

if __name__ == "__main__":
    vec1 = [1.0, 2.0, 3.0]
    vec2 = [4.0, 5.0, 6.0]
    print("Manhattan Distance:", manhattan_distance(vec1, vec2))
